### PR TITLE
Modify checkout controller callback #Fix 6724

### PR DESCRIPTION
**Pull request by aishmita-aggarwal**

### CODE REVIEW

File: frontend/app/controllers/spree/checkout_controller.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def correct_state
     def ensure_valid_state
       if @order.state != correct_state && !skip_state_validation?
         flash.keep
-        @order.state = correct_state
+        @order.update_column(:state, correct_state)
```
Comment (vishalzambre) [inline frontend/app/controllers/spree/checkout_controller.rb]: @aishmita-aggarwal  Don't think that update is required here.

Comment (aishmita-aggarwal) [inline frontend/app/controllers/spree/checkout_controller.rb]: @vishalzambre When we do not update state of order here, correct checkout page is rendered but order is in invalid state. For reference: issue #6724

**Comment by Mafi88** [comment]
:+1:

**Comment by damianlegawiec** [comment]
:+1:

**Pull request merged**